version: '3'
services:
# Nginxコンテナ
  nginx:
    container_name: nginx_spro
    image: nginx:alpine
    depends_on:
      - app
      - db
    environment:
      TZ: "Asia/Tokyo"
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./docker_conf/nginx/conf.d:/etc/nginx/conf.d
      - ./docker_conf/nginx/ssl:/etc/nginx/ssl
    networks:
      spro:
        ipv4_address: 172.27.1.11

# アプリケーションコンテナ
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: app_spro
    volumes:
      - .:/spro/
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - HOSTPWD=${PWD}
      - PYTHONPATH=/spro
    expose:
      - 8000
    depends_on:
      db:
        condition: service_healthy
    restart: always
    tty: true
    networks:
      spro:
        ipv4_address: 172.27.1.12


# DBコンテナ
  db:
    image: postgres:12.4-alpine
    container_name: db_spro
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=password
      - POSTGRES_DB=spro
      - POSTGRES_INITDB_ARGS=--encoding=UTF-8 --locale=C
      - PGDATA=/var/lib/postgresql/data/spro
    volumes:
      - db_data:/var/lib/postgresql/data
      - ./docker_conf/postgres/initdb:/docker-entrypoint-initdb.d
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD", "pg_isready", "-hdb", "-Uroot", "-p5432", "-dspro"]
      timeout: 5s
      retries: 10
    networks:
      spro:
        ipv4_address: 172.27.1.13

volumes:
  db_data:
    driver: local

networks:
  spro:
    ipam:
      driver: default
      config:
        - subnet: 172.27.1.0/24
